src/basic/cgroup-util.c

```diff
@@ -624,11 +624,7 @@ int cg_get_path(const char *controller, const char *path, const char *suffix, ch
         if (!cg_controller_is_valid(controller))
                 return -EINVAL;
 
-        unified = cg_all_unified();
-        if (unified < 0)
-                return unified;
-
-        if (unified > 0)
+        if (cg_all_unified())
```

Comment:
The assertion is, mostly, about internal logic and can be made fully so by adding "if (cg_unified_flush() < 0) err_and_exit;" whenever we're launching a new program. It's a state that systemd must know to operate in any capacity and which can be decided once when it starts up. Propagating error code from these functions is just noise. They don't serve any practical purpose. 

I left the lazy detection part in because I didn't know a good way to initialize this automatically and was too lazy to track down all the places where this should be initialized. I fix that up but am pretty sure that the error returns from functions which return cached results should go.